packages/block-editor/src/store/selectors.js

```diff
@@ -2957,7 +2960,21 @@ export const getBlockEditingMode = createRegistrySelector(
 			}
 			const rootClientId = getBlockRootClientId( state, clientId );
 			const templateLock = getTemplateLock( state, rootClientId );
-			if ( templateLock === 'contentOnly' ) {
+			const sectionRootClientId = getSectionRootClientId( state );
+			const sectionsClientIds = getBlockOrder(
+				state,
+				sectionRootClientId
+			);
+			if (
+				templateLock === 'contentOnly' ||
+				editorMode === 'navigation'
+			) {
+				// Sections should always be contentOnly in navigation mode.
```

Comment:
I'm also wondering if the `sectionClientIds` logic should be mixed with the `templateLock` logic. Could there ever be a situation where the section root has `templateLock: 'contentOnly'` and then the sections logic unexpectedly applies? Not sure, maybe it's nothing to worry about 😄

Reply:
I don't understand Dave's question.
For the latter @talldan The sections root (parent of the sections) has a special case above.

Reply:
I think Dave and I are talking about the same thing (but I may be wrong). What I mean is when you could have this situation with the `editorMode` set to `edit`:
- Group A (Section root, has `templateLock: 'contentOnly'`)
  - Group B
    - Paragraph
  
If we're calling `getBlockEditingMode` for 'Group B', then we enter the body of this `if` statement, and because Group B is listed in `sectionsClientIds` it would incorrectly be set to `contentOnly` mode.

I get that it's a really contrived example though (and I'm not sure exactly what blocks are section roots).

I guess a simple way to handle it would be to change the check below to `if ( sectionsClientIds.includes( clientId ) && editorMode === 'navigation' ) {`, because think those sections should only be active in navigation mode.